pkg/reconciler/route/route.go

```diff
@@ -480,10 +447,92 @@ func (c *Reconciler) ensureFinalizer(route *v1alpha1.Route) error {
 	return err
 }
 
+func (c *Reconciler) updateRouteStatusURL(ctx context.Context, route *v1alpha1.Route) error {
+	serviceNames, err := c.getServiceNames(ctx, route)
+	if err != nil {
+		return err
+	}
+
+	mainRouteServiceName, err := domains.HostnameFromTemplate(ctx, route.Name, "")
+	if err != nil {
+		return err
+	}
+
+	mainRouteMeta := route.ObjectMeta.DeepCopy()
+	labels.SetVisibility(mainRouteMeta, serviceNames.clusterLocal().Has(mainRouteServiceName))
+
+	host, err := domains.DomainNameFromTemplate(ctx, *mainRouteMeta, route.Name)
+	if err != nil {
+		return err
+	}
+
+	route.Status.URL = &apis.URL{
+		Scheme: "http",
+		Host:   host,
+	}
+
+	return nil
+}
+
+func (c *Reconciler) getServiceNames(ctx context.Context, route *v1alpha1.Route) (serviceNames, error) {
+	// Populate existing service name sets
+	existingServices, err := c.getServices(route)
```

Comment:
I'd return *serviceNames, but it seems small enough to not to care

Reply:
I think probably they can be computed even before, but if something isn't used -- no need to have it around.

Reply:
This was for the other comment :-)